docs/features/network-security-controls/network-policy.md

```diff
@@ -58,7 +178,107 @@ spec:
   policyTypes:
   - Egress
 ```
-**OVN-Implementation:**
+
+#### **Selector Based Denial**
+
+We can extend the default denial behavior by using the `podSelector` field to selectively block all traffic from/to certain pods, in this case blocking all unicast traffic from reaching apps labeled with demo.
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: default-deny-all
+spec:
+  podSelector:
+    matchLabels:
+          app: demo
+  policyTypes:
+  - Ingress
+  - Egress
+```
+
+#### **Selector Based filtering**
+
+The `from` selector for an ingress and the `to` selector for an egress can be used to select particular pods or namespaces that will be allowed as destinations or sources of data, as in the example below, which allows all incoming traffic from the namespace db_app into the networkPolicy namespace.
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-db-source
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - namespaceSelector:
+        matchLabels:
+        name: db_app
+```
+
+
+#### **CIDR Based Filtering**
+
+The `from` selector for an ingress and the `to` selector for an egress can also be used to specify specific CIDR ranges or `ip blocks`, allowing developers to create granular ACLs. In the example below, we allow all source connections from the 10.16.3.0/24 subnet that are communicating on the mySQL port tcp/3306
+
+```
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: CIDR-db-app
+spec:
+  podSelector: {}
+  policyTypes:
+  - Ingress
+  ingress:
+  - from:
+    - ipBlock:
+        cidr: 10.16.3.0/24
+        except:
+        - 10.16.3.1
+    ports:
+    - protocol: tcp
+      port: 3306
+```
+
+#### **Complex Filtering**
+
+Each egress/ingress rule is designed to be combined with other rules to create more complex filters. These will get applied in order of precendence, with the first rule being applied first. In this example the application allows ingress traffic from both the db_namespace and content servers located in the 10.16.24.0/24 subnet while only allowing egress through 173.23.32.4/32
```

Comment:
there is no precedence in NetworkPolicy, all allow rules have the same priority, and everything that is not allowed will be denied